### PR TITLE
Fix check-mod-tidy invoke task rerun

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR add the removal of the `testmodule` folder at the end of the `check-mod-tidy` task.

### Motivation

The `check-mod-tidy` invoke task fails (`FileExistsError: [Errno 17] File exists: 'testmodule`) if there's already a `testmodule` folder in the current folder (as the result of a previous `check-mod-tidy` run), making it annoying to run the task multiple times.

### Describe how to test/QA your changes

You can run `inv -e check-mod-tidy` several times in a row without any error.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/go.py [hunk 1]
```diff
@@ -490,6 +490,9 @@ def check_mod_tidy(ctx, test_folder="testmodule"):
         message += "\n\nRun 'inv tidy-all' to fix 'out of sync' errors."
         raise Exit(message=message)
 
+    # delete test_folder to avoid FileExistsError while running this task again
+    ctx.run(f"rm -rf ./{test_folder}")
```
[the remark this answers is not in the document]
Comment (amenasria) [reply tasks/go.py]: Missed that, thank you !

File: tasks/go.py [hunk 2]
```diff
@@ -485,13 +485,14 @@ def check_mod_tidy(ctx, test_folder="testmodule"):
...
             os.remove(os.path.join(ctx.cwd, "main"))
 
-    if errors_found:
-        message = "\nErrors found:\n" + "\n".join("  - " + error for error in errors_found)
-        message += "\n\nRun 'inv tidy-all' to fix 'out of sync' errors."
-        raise Exit(message=message)
-
-    # delete test_folder to avoid FileExistsError while running this task again
-    ctx.run(f"rm -rf ./{test_folder}")
+    try:
```
Comment (albertvaka) [inline tasks/go.py:488]: Maybe move this up till the line the directory is created (`generate_dummy_package`).

Comment (albertvaka) [reply tasks/go.py:488]: Or even cooler if you want to look into it, we could implement a `with` block like:
```
with generate_dummy_package(ctx, test_folder):
   ...
```
That deletes the folder at the end regardless of exceptions.

File: tasks/go.py [hunk 3]
```diff
@@ -468,27 +468,27 @@ def reset(ctx):
...
+            with ctx.cd(mod.full_path()):
+                ctx.run("go mod tidy -compat=1.17")
+                res = ctx.run("git diff-files --exit-code go.mod go.sum", warn=True)
+                if res.exited is None or res.exited > 0:
+                    errors_found.append(f"go.mod or go.sum for {mod.import_path} module is out of sync")
+
+        with ctx.cd(dummy_folder):
             ctx.run("go mod tidy -compat=1.17")
-            res = ctx.run("git diff-files --exit-code go.mod go.sum", warn=True)
+            res = ctx.run("go build main.go", warn=True)
```
Comment (mx-psi) [inline tasks/go.py:482]: I don't get this change 🤔

**Review by mx-psi — COMMENTED** [review]
LGTM except for one detail I don't understand

**Review by mx-psi — APPROVED** [review]

**Pull request merged**